tests/TestCase/src/Controller/BlogPostsControllerTest.php

```diff
@@ -29,41 +32,43 @@ public function setUp(): void
 
     public function testBlogPostDetailOnlinePublicLoggedOut()
     {
-        $this->httpClient->get($this->Slug->getBlogPostDetail(2, 'Demo Blog Artikel'));
-        $this->assert200OkHeader();
+        $_SERVER['REQUEST_URI'] = $this->Slug->getBlogPostDetail(2, 'Demo Blog Artikel');
+        $this->get($_SERVER['REQUEST_URI']);
+        $this->assertResponseCode(200);
     }
 
     public function testBlogPostDetailOfflinePublicLoggedOut()
     {
         $blogPostId = 2;
         $this->changeBlogPost($blogPostId, 0, 0, 0);
-        $this->httpClient->get($this->Slug->getBlogPostDetail($blogPostId, 'Demo Blog Artikel'));
-        $this->assert404NotFoundHeader();
+        $this->get($this->Slug->getBlogPostDetail($blogPostId, 'Demo Blog Artikel'));
+        $this->assertResponseCode(404);
     }
 
     public function testBlogPostDetailOnlinePrivateLoggedOut()
     {
         $blogPostId = 2;
         $this->changeBlogPost($blogPostId, 1);
-        $this->httpClient->followOneRedirectForNextRequest();
-        $this->httpClient->get($this->Slug->getBlogPostDetail($blogPostId, 'Demo Blog Artikel'));
-        $this->assertAccessDeniedWithRedirectToLoginForm();
+        $this->get($this->Slug->getBlogPostDetail($blogPostId, 'Demo Blog Artikel'));
+        $this->Slug->getLogin($blogPostId, 'Demo Blog Artikel');
+        $this->assertFlashMessage('Zugriff verweigert, bitte melde dich an.');
     }
 
     public function testBlogPostDetailOnlinePrivateLoggedIn()
     {
         $this->loginAsCustomer();
         $blogPostId = 2;
         $this->changeBlogPost($blogPostId, 1);
-        $this->httpClient->get($this->Slug->getBlogPostDetail($blogPostId, 'Demo Blog Artikel'));
-        $this->assert200OkHeader();
+        $_SERVER['REQUEST_URI'] = $this->Slug->getBlogPostDetail($blogPostId, 'Demo Blog Artikel');
```

Comment:
Please use option 1 (define server-element in phpunit.dist.xml)
https://github.com/foodcoopshop/foodcoopshop/pull/529#discussion_r431565459